appserver/tests/payara-samples/classloader-data-api/src/main/java/fish/payara/samples/classloaderdata/InstanceCountTracker.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package fish.payara.samples.classloaderdata;
+
+import java.lang.ref.Reference;
+import org.glassfish.web.loader.WebappClassLoader;
+import org.glassfish.web.loader.WebappClassLoaderFinalizer;
+
+/**
+ *
+ * @author Cuba Stanley
+ */
+public class InstanceCountTracker {
```

Comment:
I would move instance count into own class, not have instance counts in different classes.
Minimize changes to WebappClassLoader

Also, this doesn't touch non-webapp (jar, ear etc) class loaders